src/module.c

```diff
@@ -504,6 +508,37 @@ void *RM_PoolAlloc(RedisModuleCtx *ctx, size_t bytes) {
  * Helpers for modules API implementation
  * -------------------------------------------------------------------------- */
 
+client *moduleAllocTempClient()
+{
+    client *c;
+
+    pthread_mutex_lock(&moduleTempClientsMutex);
+    if (moduleTempClientCount > 0) {
+        c = moduleTempClients[--moduleTempClientCount];
+    } else {
+        c = createClient(NULL);
+    }
+    pthread_mutex_unlock(&moduleTempClientsMutex);
+    return c;
+}
+
+void moduleReleaseTempClient(client *c)
+{
+    const size_t cap = sizeof(moduleTempClients) / sizeof(moduleTempClients[0]);
+
+    pthread_mutex_lock(&moduleTempClientsMutex);
+    if (moduleTempClientCount == cap) {
+        freeClient(c);
+    } else {
+        listEmpty(c->reply);
+        c->reply_bytes = 0;
+        resetClient(c);
+        c->bufpos = 0;
```

Comment:
What else should we clear here? I assumed these clients are used for accumulating replies so clearing replies only. Is there any other use case for these temp clients?

Reply:
maybe we need to extract some code from `resetCommand` and re-use it here?
maybe `moduleFreeContextReusedClient` should also use it?